qxf2_scheduler/qxf2_scheduler.py

```diff
@@ -38,7 +50,83 @@ def is_weekend(date):
     date = gcal.process_date_string(date)
     day = date.weekday()
     print('~~~~',day)
-    return True if day==5 or day==6 else False 
+    return True if day==5 or day==6 else False
+
+
+def get_free_slots_in_chunks(free_slots):
+    "Return the free slots in 30 minutes interval"
+    #Appending the 30 minutes slot into list
+    if free_slots == None:
+        print("There are no more free slots available for this user")    
+    else:        
+        chunk_time_interval = []        
+        for i in  range(0,len(free_slots)):
+            #Initializing the free slot end            
+            free_slot_start = free_slots[i]['start']
+            #Intializing the next free slot start        
+            free_slot_end = free_slots[i]['end']
+            
+            #Find the difference between start and end slot
+            diff_between_slots = convert_string_into_time(free_slot_end) - convert_string_into_time(free_slot_start)
+            
+            if diff_between_slots >= timedelta(minutes=30):                
+                if free_slot_start[-2:]=='00':
```

Comment:
Can you make lines 73-93 into two methods. That way, we can simply do:

free_slot_start = get_modified_free_slot_start(free_slots[i]['start'],marker='30')
free_slot_end = get_modified_free_slot_end(free_slots[i]['end'],marker='30')

Reply:
Done